en/middleware/micrortps.md

```diff
@@ -0,0 +1,261 @@
+# RTPS/ROS2 Interface: PX4-FastRTPS Bridge
+
+The *PX4-FastRTPS Bridge* adds a Real Time Publish Subscribe (RTPS) interface to PX4, enabling the exchange of [uORB messages](../middleware/uorb.md) between PX4 components and (offboard) *FastRTPS* applications.
+
+> **Note** RTPS is the underlying protocol of the Object Management Group's (OMG) Data Distribution Service (DDS) standard. It aims to enable scalable, real-time, dependable, high-performance and interoperable data communication using the publish/subscribe pattern. *FastRTPS* is a very lightweight cross-platform implementation of the latest version of the RTPS protocol and a minimum DDS API.
+
+RTPS has been adopted as the middleware for the ROS2 (Robot Operating System). The bridge allows us to better integrate with ROS2, making it easy to share sensor values, commands, and other vehicle information.
+
+This topic describes the bridge architecture, how it is compiled, and how to write a simple FastRTSP application to subscribe to PX4 changes.
+
+
+## Architectural overview
+
+![basic example flow](../../assets/middleware/micrortps/basic_example_flow.png)
+
+The main elements of the architecture are the client and agent processes shown in the diagram above.
+
+- The *Client* is PX4 middleware daemon process that runs on the flight controller. It subscribes to uORB topics published by other PX4 components and sends any updates to the *Agent* (via a UART or UDP port). It also receives messages from the *Agent* and publishes them as uORB message on PX4.
```

Comment:
From a docs point of view it does not matter - as a user of this doc you just need to know that it is a "uORB" message. Additional detail would live in the "Manual generation" doc - but I question whether more detail is useful even there. 

That said, I am 90% sure they are a CDR serialized version of the uORB message, and that the further conversion to RTPS format happens in the agent.